File: app/src/main/java/br/com/jaysonsabino/desafioandroidpopcode/database/AppDatabase.java
package br.com.jaysonsabino.desafioandroidpopcode.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import br.com.jaysonsabino.desafioandroidpopcode.entities.Character;
import br.com.jaysonsabino.desafioandroidpopcode.entities.FavoriteCharacter;

@Database(entities = {Character.class, FavoriteCharacter.class}, version = 5, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    public abstract CharacterDAO getCharacterDAO();

    public abstract FavoriteCharacterDAO getFavoriteCharacterDAO();
}

File: app/src/main/java/br/com/example/starwars/data/local/AppDataBase.kt
package br.com.example.starwars.data.local

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import br.com.example.starwars.data.entities.ApiPeople
import br.com.example.starwars.data.entities.RemoteKeys
import br.com.example.starwars.data.util.StringListConverter

@Database(entities = [ApiPeople::class, RemoteKeys::class], version = 1, exportSchema = false)
@TypeConverters(StringListConverter::class)
abstract class AppDataBase : RoomDatabase() {

    abstract fun peopleDao(): PeopleDao
    abstract fun remoteKeysDao(): RemoteKeysDao

    companion object {
        private const val DATABASE_NAME = "star-wars-db"

        fun build(context: Context): AppDataBase {
            return Room.databaseBuilder(
                context,
                AppDataBase::class.java, DATABASE_NAME
            ).build()
        }
    }
}